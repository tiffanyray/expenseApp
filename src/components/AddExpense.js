import React from 'react';

const AddExpense = () => (
    <div>
        <h1>This is my add expense component</h1>
    </div>
);

export default AddExpense;