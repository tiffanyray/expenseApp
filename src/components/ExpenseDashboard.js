import React from 'react';

const ExpenseDashboardPage = () => (
    <div>
        <h1>This is from dashboard component</h1>
    </div>
);

export default ExpenseDashboardPage;