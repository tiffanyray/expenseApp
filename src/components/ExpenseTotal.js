import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenseTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';


export const ExpenseTotal = ({ expenseCount, expenseTotal }) => {
    const formattedExpensesTotal = numeral(expenseTotal).format('$0,0.00');
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';

    return (
        <div>
            <p>You are viewing {expenseCount} {expenseWord} that total {formattedExpensesTotal}</p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpenseTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseTotal);