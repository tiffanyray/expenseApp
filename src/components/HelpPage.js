import React from 'react';

const HelpPage = () => (
    <div>
        <h1>This is my help component</h1>
    </div>
);

export default HelpPage;