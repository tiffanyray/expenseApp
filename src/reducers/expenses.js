

// Expenses Reducer 
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state,
        action.expense
      ]
    case 'REMOVE_EXPENSE' :
      return state.filter( ({ id }) => id !== action.id);
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if(expense.id === action.id) {
          return { 
            ...expense, //spread operator that makes it easy to 
            ...action.updates //update any of the existing infomation provided
          }
        } else {
          return expense;
        }
      })
    default: 
      return state;
  }
};

export default expensesReducer;