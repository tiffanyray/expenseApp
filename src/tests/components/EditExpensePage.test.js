import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(
        <EditExpensePage 
            editExpense={editExpense} 
            removeExpense={removeExpense} 
            history={history}
            expense={expenses[2]}
        />)
}) 
//getting initial render snapshot of component
test('should render initial snaphot of editexpensepage', () => {
    expect(wrapper).toMatchSnapshot();
});

//handles edit expense using spies
test('should handle spies edit expense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[2]);
});

//handles remove expense using spies
test('should handle remove expense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({
        id: expenses[2].id
    });

});