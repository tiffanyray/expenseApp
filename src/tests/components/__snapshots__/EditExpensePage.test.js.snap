// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial snaphot of editexpensepage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "credit card",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove Item
  </button>
</div>
`;
